packages/react/src/menu/submenu-trigger/useMenuSubmenuTrigger.ts

```diff
@@ -32,17 +41,37 @@ export function useMenuSubmenuTrigger(
 
   const menuTriggerRef = useForkRef(menuItemRef, setTriggerElement);
 
+  const direction = useDirection();
+
+  const [isSubmenuOpen, setIsSubmenuOpen] = useState(false);
+
   const getRootProps = React.useCallback(
     (externalProps?: GenericHTMLProps) => {
-      return {
+      const openKey = direction === 'rtl' ? 'ArrowLeft' : 'ArrowRight';
+      const handleOpenSubmenu = () => {
+        if (highlighted) {
+          setActiveIndex(null);
+          setIsSubmenuOpen(true);
+        }
+      };
+      return mergeReactProps(externalProps, {
```

Comment:
`mergeReactProps<'div'>(...)` will be able to type the `onKeyDown` event automatically